connectors/database/postgres.go

```diff
@@ -0,0 +1,39 @@
+package sql
+
+import (
+	"fmt"
+
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+func InitPostgresClient(dbConfig map[string]string) *gorm.DB {
+	host := dbConfig["host-v3"]
+	username := dbConfig["username-v3"]
+	password := dbConfig["password-v3"]
+	port := dbConfig["port-v3"]
+	dbname := dbConfig["database-v3"]
+
+	if host == "" {
+		return nil
+	}
+
+	dsn := fmt.Sprintf("host=%s user=%s password=%s dbname=%s port=%s", host, username, password, dbname, port)
```

Comment:
Add validation for the presence of all required keys in the `dbConfig` map before constructing the `dsn`. This can prevent runtime errors and ensure that the function behaves predictably. For example, check if `username`, `password`, `port`, and `dbname` are empty and handle these cases appropriately. [important]